application/backend/eye_tracker_newsdk.py

```diff
@@ -298,6 +264,36 @@ def on_gazedata_4c(self, x, y, time_stamp):
         self.validity.append(True)
         self.LastTimestamp = time_stamp
         self.dpt_id += 1
+		
+    def on_gazedata_simulation(self, x, y, time_stamp):
```

Comment:
interesting I just realize that we have these on_gaze_data(), on_gaze_data_4c() and on_gazedata_simulation() functions.
I wonder if we could move them to eye_tracker_class? Like add a new on_gaze_data() to each tracker?

Reply:
seems like there are lots of variables being set inside these functions so I will have to take a closer look

Reply:
okay, there are 2 ways I can think of refactoring this:

Either way I can refactor on_gaze_data to be specific to each tracker

1. I refactor/rename all instances where the variables inside `on_gaze_data` were accessed from `self.tobii_controller.<variable_name>` to `self.tobii_controller.eye_tracker.<variable_name>`
2. I keep all variables defined in the `TobiiControllerNewSdk` class, and after the call to the eye tracker specific `on_gaze_data` I define a function that will update the `TobiiControllerNewSdk`'s variables' values, so that nothing outside of the class will have to change

2 sounds quite messy internally to the class, but it's nice because nothing else has to change. On the other hand, 1 is more straightforward and probably cleaner in the long run (?), but it means that outside classes will have access to the specific eye tracker implementation and its variables. 
Also, I am not sure at which level of abstraction the eye tracker stuff should be; should a class outside of the TobiiControllerNewSdk really be able to access its implementation of the eye tracker and its variables?

Reply:
let's keep it for another round of work/PR, I need to think a bit more about your two options and wrap my head around them :)